### ISSUE TITLE
Automatically detect the QEMU disk

### ISSUE THREAD

**Issue by gdamjan** [gdamjan/archvm-auto-install#3]
Depending on the device emulation selected for the virtual disk, the target disk can appear as `/dev/sda`, `/dev/vda` or even any other letter in the guest. Currently the target disk is hard-coded to `/dev/sda`, but that is fragile.

One option would be to invoke qemu with:
```
-device virtio-blk-pci,drive=hd0,serial=$DISK_SERIAL
-smbios "type=11,value=root-disk-serial:$DISK_SERIAL"
```

so that we configure the disk with a serial number, and also pass that serial as a smbios/dmi data.
The install script can then take the serial and use lsblk to find the matching block device.

For ex.
```
DISK_SERIAL=`strings /sys/firmware/dmi/entries/11-0/raw | sed -n '/^root-disk-serial:/s/^root-disk-serial://p'`
SYSDISK=`lsblk -JO | jq -r '.blockdevices[] | select(.serial == "'$DISK_SERIAL'") | .path'`
```
(one problem is, ArchIso doesn't have `jq` installed by default)

note,
  not all disk emulations in qemu support the `serial` parameter (afaik only `virtio-blk-pci`, `scsi-hd` and `ide-hd` do)

**Comment by gdamjan** [comment]
Another option is
```
-fw_cfg name=opt/root-disk-serial,string=$DISK_SERIAL`
```
```
DISK_SERIAL=`< /sys/firmware/qemu_fw_cfg/by_name/opt/root-disk-serial/raw`
```

**Comment by gdamjan** [comment]
fixed in [907df01](https://github.com/gdamjan/archvm-auto-install/commit/907df012b82916dd211b4ead03499c1ad9fdc9a8)
